accelerator/http.c

```diff
@@ -360,8 +360,9 @@ static int ta_http_handler(void *cls, struct MHD_Connection *connection, const c
   // While upload_data_size > 0 process upload_data
   if (*upload_data_size > 0) {
     if (http_req->request == NULL) {
-      http_req->request = (char *)malloc(*upload_data_size);
+      http_req->request = (char *)malloc((*upload_data_size) + 1);
```

Comment:
Shall we validate the return value of `malloc`? In fact, there is no mechanism to check requested size in advance.